src/ode_training.py

```diff
@@ -236,11 +239,28 @@ def main(config):
         parameter_vector = Variable(torch.from_numpy(parameter_vector).float(), requires_grad=True).to(device)
 
         # Loss based on CRT ODEs
-        residual = ode_equation.compute_ode_residual(x_flux_vector, x_state_vector, x_time_vector, parameter_vector, u_approximation)
-        log_residual = torch.log10(torch.abs(residual))
-        # loss: [log(residual) - precision]^2
-        loss_ode = F.mse_loss(input=log_residual, target=target_residual, reduction='mean')
-
+        r_x_H_II, r_x_He_II, r_x_He_III, r_T = ode_equation.compute_ode_residual(x_flux_vector,
+                                                                                       x_state_vector,
+                                                                                       x_time_vector,
+                                                                                       parameter_vector,
+                                                                                       u_approximation)
+
+        # [Issue] using log10(abs(prediction)) here introduces two problems:
+        # 1. we lose sign information
+        # 2. log doesn't have a minimum so we cant minimise it.
+        # possible fixes: use sigmoid or tanh here, which helps us get rid of
+        # both above issues
+        out_x_H_II = torch.tanh(r_x_H_II)
```

Comment:
We should probably discuss this in the next meeting

Reply:
@eor Yeah sure. I did this and found the curves to make a bit more sense. Although I think there are a lot of other issues which need to be addressed first before comparing the scaling functions that can be used here.